### DISCUSSION TITLE
Problem using the "mkdocs-nav-weight" plugin - How to implement nav without hard-coding?

### DISCUSSION THREAD

**Discussion by eve-archivist** [squidfunk/mkdocs-material#6049] [Q&A] [answered]
I am trying to solve the problem of not wanting to hard-code site navigation using `nav:` as I am building a creative fiction site where content are constantly added.  It would be more maintainable if the sidebar navigation tree is autogenerated, but still be to specify an order that is not alphabetical but based on front-matter attributes.

I thought I found the solution with shu307's `mkdocs-nav-weight`, this is similar the solution provided by Hugo's Docsy theme from Google, but as soon as I include the plugin in my mkdocs.yml:
    
```
    plugins:
      - mkdocs-nav-weight
```

I get an error with previewing:

```
    % mkdocs serve
     ERROR   -  Config value 'plugins': Invalid Plugins configuration
     Aborted with 1 configuration errors!
```

If I comment out `material` theme and run it under vanilla mkdocs, everything works as expected with tabs/sidebar sorted by page weight.

I find Material contains a schema/plugins.json which is referenced by schema.json for syntax validation, does this also prevent "unknown" plugins to be flagged with run-time error?

Or maybe it is my own user error?

Btw, really like the mkdocs + material package.  I tried out different alternatives last few days, and this combination is by far the most flexible and easy to use!

**Comment by alexvoss** [comment]
I have tried but failed to reproduce this with a project I have. Adding the `mkdocs-nav-weight` at the same time as having the `material` theme works for me and I can control the order of the pages using weights. Can you please produce a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/) for this?

NB: I don't think this has anything to do with the schema. I used `grep` to search for references to it in the source code and found none - outside the schema files themselves, their documentation and the VS Code configuration. I know the schema does not support additional entries and I wonder if that should be changed but I don't think this is the problem here.

**Reply by eve-archivist** [reply, marked as answer]
Thanks for trying to reproduce my problem.

It was my noob user mistake.

```
site_name: Test
plugins:
    - literate-nav
    - mkdocs-nav-weight:
      section_renamed: false
theme:
   name: material
```
need to indent two more space at line 5!!

[FacePalm]

**Reply by alexvoss** [reply]
We've all been there! Glad you figured it out.

**Reply by eve-archivist** [reply]
As part of this exercise, I discovered the plugin `mkdocs-literate-nav`, which also do some very nice thing.

Now I wonder if I can use both plugin together, and what the use cases might be for that!

My sincere apology for not testing more thoroughly before posting.  

I was also confused by VS Code syntax checking with after setup with YAML + schema.jason.

It will flag this as invalid in the editor
![mkdocs_yml](https://github.com/squidfunk/mkdocs-material/assets/145245633/5392651b-eba0-45ab-b838-51cad8932b5b)

But would accept this:
![mkdocs_yml2](https://github.com/squidfunk/mkdocs-material/assets/145245633/a058e723-21cb-41f4-b0b7-4d10eae841a3)

Both actually works, of course, just the editor syntax checking.
